pkg/actuators/machine/vm.go

```diff
@@ -14,6 +15,146 @@ import (
 	clientpkg "github.com/openshift/machine-api-provider-nutanix/pkg/client"
 )
 
+// validateVMConfig verifies if the machine's VM configuration is valid
+func validateVMConfig(mscp *machineScope) field.ErrorList {
+	errList := field.ErrorList{}
+	fldPath := field.NewPath("spec", "providerSpec", "value")
+
+	var err error
+	var errMsg string
+
+	// verify the cluster configuration
```

Comment:
Consider Breaking down the validation logic into submethods (i.e. validateClusterConfig , validateImageConfig, validateSubnetConfig)